feat(curriculum): add string method workshop (#66936)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-employee-profile-validator/69e62ce96df237884da25282.md

---
id: 69e62ce96df237884da25282
title: Step 9
challengeType: 20
dashedName: step-9
---

# --description--

> The `.lower()` method converts all characters in a string to lowercase. This is useful for normalising email addresses:

Reviewer: ```suggestion
The `.lower()` method converts all characters in a string to lowercase. This is useful for normalizing email addresses:
```

```python
email = '  HELLO@EXAMPLE.COM  '
clean = email.strip().lower()
print(clean)  # Output: hello@example.com
```

Create a variable `messy_email` and assign it `'  SARAH.DAVIS@COMPANY.COM  '`.

Then create `clean_email` by calling `.strip().lower()` on `messy_email`.

Then print both like this:

```python
print('Original email: '' + messy_email + ''')

> print('Cleaned email: '' + clean_email + ''')

Reviewer 2: Are you sure? I believe this is a failed search and replace

Reviewer 3: Hope that's fine now

```

# --hints--

You should create a variable named `messy_email`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('messy_email')`)) })
```

You should assign `'  SARAH.DAVIS@COMPANY.COM  '` to `messy_email`.

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('messy_email')
not node.is_empty() and node.is_equivalent("messy_email = '  SARAH.DAVIS@COMPANY.COM  '")
`)) })
```

You should create a variable named `clean_email`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('clean_email')`)) })
```

You should use `.strip().lower()` on `messy_email` to assign `clean_email`.

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('clean_email')
not node.is_empty() and node.is_equivalent("clean_email = messy_email.strip().lower()")
`)) })
```

You should print the original email.

```js

> ({ test: () => assert(runPython(`_Node(_code).has_call("print('Original email: ' + messy_email + \"'\")")`)) })

Reviewer: ```suggestion
({ test: () => assert(runPython(`_Node(_code).has_call("print(\\"Original email: '\\" + messy_email + \\"'\\")")`)) })
```

```

You should print the cleaned email.

```js

> ({ test: () => assert(runPython(`_Node(_code).has_call("print('Cleaned email: ' + clean_email + \"'\")")`)) })

Reviewer: ```suggestion
({ test: () => assert(runPython(`_Node(_code).has_call("print(\\"Cleaned email: '\\" + clean_email + \\"'\\")")`)) })
```

---
id: 69e62ce96df237884da25282
title: Step 10
challengeType: 20
dashedName: step-10
---

# --description--

The `.lower()` method converts all characters in a string to lowercase. This is useful for normalizing email addresses:

```python
email = '  HELLO@EXAMPLE.COM  '
clean = email.strip().lower()
print(clean)  # Output: hello@example.com

> ```

Reviewer 4: This example names its variable `email`, which is already a variable in the camper's locked seed holding `'alice.johnson@company.com'`. Renaming it avoids the collision, and the fence matches the rest of the workshop.

`````suggestion
```py
messy = '  HELLO@EXAMPLE.COM  '
clean = messy.strip().lower()
print(clean)  # Output: hello@example.com
```
`````

Create a variable `messy_email` and assign it `'  SARAH.DAVIS@COMPANY.COM  '`.

Then create `clean_email` by calling `.strip().lower()` on `messy_email`.

Then print both like this:

> ```python

Reviewer 4: ````suggestion
```py
````

print("Original email: '" + messy_email + "'")
print("Cleaned email: '" + clean_email + "'")
```

# --hints--

You should create a variable named `messy_email`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('messy_email')`)) })
```

You should assign <code>'&nbsp;&nbsp;SARAH.DAVIS@COMPANY.COM&nbsp;&nbsp;'</code> to `messy_email`.

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('messy_email')
not node.is_empty() and node.is_equivalent("messy_email = '  SARAH.DAVIS@COMPANY.COM  '")
`)) })
```

You should create a variable named `clean_email`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('clean_email')`)) })
```

> You should use `.strip().lower()` on `messy_email` to assign a value to `clean_email`.

Reviewer 4: ```suggestion
You should use `.strip().lower()` on `messy_email` to assign `clean_email` a value.
```

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('clean_email')
not node.is_empty() and node.is_equivalent("clean_email = messy_email.strip().lower()")
`)) })
```

You should print the original email with the prefix `"Original email: '"`. Don't forget to add the closing single quote.

```js
({ test: () => assert(runPython(`_Node(_code).has_call("print(\\"Original email: '\\" + messy_email + \\"'\\")")`)) })
```

> You should print the cleaned email with the prefix "Cleaned email: '". Don't forget to add the closing single quote.

Reviewer 4: The prefix is backticked in the preceding hint but bare here.

```suggestion
You should print the cleaned email with the prefix `"Cleaned email: '"`. Don't forget to add the closing single quote.
```

---
id: 69e62ce96df237884da25282
title: Step 10
challengeType: 20
dashedName: step-10
---

# --description--

The `.lower()` method converts all characters in a string to lowercase. This is useful for normalizing email addresses:

```python
upperGreet = 'HELLO WORLD'
lowerGreet = upperGreet.lower()

> print(lowerGreet)  # Output: hello world

Reviewer 4: `upperGreet` and `lowerGreet` are the only camelCase names anywhere in the workshop, and PEP 8 asks for snake_case. Every other example variable here (`messy`, `clean`, `parts`, `joined`, `text`, `code`, `phone`) already follows it.

```suggestion
upper_greet = 'HELLO WORLD'
lower_greet = upper_greet.lower()
print(lower_greet)  # Output: hello world
```

```

You can call both `.strip()` and `.lower()` on a string this way:

```python
messy = '  HELLO WORLD  '
clean = messy.strip().lower()
print(clean)  # Output: hello world
```

Create a variable `messy_email` and assign it the string `  SARAH.DAVIS@COMPANY.COM  `. Pay attention to the two leading spaces and the two trailing spaces.

> Also, call `.strip().lower()` on `messy_email` and assign the result to a variable `clean_email`.

Reviewer 4: "named" on L26, see the step 7 note.

L28 opens with "Also," where the parallel sentence in every other multi-part step opens with "Then".

```suggestion
Create a variable named `messy_email` and assign it the string `  SARAH.DAVIS@COMPANY.COM  `. Pay attention to the two leading spaces and the two trailing spaces.

Then, call `.strip().lower()` on `messy_email` and assign the result to a variable named `clean_email`.
```

Then print both so that the output reads:

```md
Original email: '  SARAH.DAVIS@COMPANY.COM  '
Cleaned email: 'sarah.davis@company.com'
```

# --hints--

You should create a variable named `messy_email`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('messy_email')`)) })
```

> You should assign <code>'&nbsp;&nbsp;SARAH.DAVIS@COMPANY.COM&nbsp;&nbsp;'</code> to `messy_email`.

Reviewer 4: Same `&nbsp;` versus literal-space split as step 7: L68 of this file writes a value with literal spaces inside a code span.

```suggestion
You should assign `'  SARAH.DAVIS@COMPANY.COM  '` to `messy_email`.
```

---
id: 69e62ce96df237884da25282
title: Step 10
challengeType: 20
dashedName: step-10
---

# --description--

The `.lower()` method converts all characters in a string to lowercase. This is useful for normalizing email addresses:

> 

Reviewer 2: Missing example code block

Create a variable `messy_email` and assign it `'  SARAH.DAVIS@COMPANY.COM  '`.

> Then create `clean_email` by calling `.strip().lower()` on `messy_email`.

Reviewer 2: ```suggestion
Create a variable `messy_email` and assign it the string `  SARAH.DAVIS@COMPANY.COM  `. Pay attention to the two leading spaces and the two trailing spaces.

Also, call `.strip().lower()` on `messy_email` and assign the result to a variable `clean_email`.
```

Then print both like this:

```python
print("Original email: '" + messy_email + "'")
print("Cleaned email: '" + clean_email + "'")
```

# --hints--

You should create a variable named `messy_email`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('messy_email')`)) })
```

You should assign <code>'&nbsp;&nbsp;SARAH.DAVIS@COMPANY.COM&nbsp;&nbsp;'</code> to `messy_email`.

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('messy_email')

> not node.is_empty() and node.is_equivalent("messy_email = '  SARAH.DAVIS@COMPANY.COM  '")

Reviewer 2: ```suggestion
_Node(_code).find_variable('messy_email').is_equivalent("messy_email = '  SARAH.DAVIS@COMPANY.COM  '")
```

`)) })
```

You should create a variable named `clean_email`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('clean_email')`)) })
```

> You should use `.strip().lower()` on `messy_email` to assign `clean_email` a value.

Reviewer 2: ```suggestion
You should call `.strip().lower()` on `messy_email` and assign the result to a `clean_email` variable.
```

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('clean_email')
not node.is_empty() and node.is_equivalent("clean_email = messy_email.strip().lower()")
`)) })
```

> You should print the original email with the prefix `"Original email: '"`. Don't forget to add the closing single quote.

Reviewer 2: ```suggestion
You should have `print("Original email: '" + messy_email + "'")` in your code.
```

```js
({ test: () => assert(runPython(`_Node(_code).has_call("print(\\"Original email: '\\" + messy_email + \\"'\\")")`)) })
```

> You should print the cleaned email with the prefix `"Cleaned email: '"`. Don't forget to add the closing single quote.

Reviewer 2: ```suggestion
You should have `print("Cleaned email: '" + clean_email + "'")` in your code.

---
id: 69e62ce96df237884da25282
title: Step 7
challengeType: 20
dashedName: step-7
---

# --description--

The `.lower()` method converts all characters in a string to lowercase. This is useful for normalising email addresses:

```python
email = '  HELLO@EXAMPLE.COM  '
clean = email.strip().lower()
print(clean)  # Output: hello@example.com
```

Create a variable `messy_email` and assign it `'  SARAH.DAVIS@COMPANY.COM  '`.

Then create `clean_email` by calling `.strip().lower()` on `messy_email`.

Then print both like this:

```python
print('Original email: '' + messy_email + ''')
print('Cleaned email: '' + clean_email + ''')
```

# --hints--

You should create a variable named `messy_email`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('messy_email')`)) })
```

You should assign `'  SARAH.DAVIS@COMPANY.COM  '` to `messy_email`.

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('messy_email')

> not node.is_empty() and (
>   node.is_equivalent('messy_email = '  SARAH.DAVIS@COMPANY.COM  '')
> )
> `)) })

Reviewer: ```suggestion
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('messy_email')
not node.is_empty() and node.is_equivalent("messy_email = '  SARAH.DAVIS@COMPANY.COM  '")
`)) })
```
